data_catalog_backend/routes/provider_routes.py

```diff
@@ -0,0 +1,66 @@
+import logging
+from typing import List
+import uuid
+from fastapi import APIRouter, Depends, HTTPException
+
+from data_catalog_backend.dependencies import get_provider_service
+from data_catalog_backend.models import Provider
+from data_catalog_backend.schemas.provider import ProviderRequest, ProviderResponse
+from data_catalog_backend.services.provider_service import ProviderService
+
+
+router = APIRouter()
+
+
+@router.post("/provider",
+             summary="Add a provider to the database",
+             tags=["admin"],
+             response_model=ProviderResponse)
+async def post_provider(provider_req: ProviderRequest,
+                        service: ProviderService = Depends(get_provider_service)
+                        ) -> ProviderResponse:
+      ed = Provider(
+            name = provider_req.name,
+            provider_url = provider_req.provider_url,
+            description = provider_req.description,
+      )
+      
+      try: 
+            created = service.create_provider(ed)
```

Comment:
Can be shortened by `created = service.create_provider(Provider(**provider_req.model_dump()))`